sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -164,6 +166,16 @@ public Object invoke(Object proxy, Method method, Object[] args) {
         + Arrays.toString(args) + "].");
   }
 
+  private void writeObject(java.io.ObjectOutputStream stream)
+      throws IOException {
+    throw new NotSerializableException(
+        "PipelineOptions objects are not serializable and should not be embedded into transforms "
+            + "(did you capture a PipelineOptions object in a field or in an anonymous class?). "
+            + "Instead, if you're using a DoFn, access PipelineOptions at runtime "
+            + "from ProcessContext.pipelineOptions(), or pre-extract necessary fields from "
```

Comment:
`ProcessContext.pipelineOptions()` -> `ProcessContext.getPipelineOptions()`

Its also available on StartBundleContext and FinishBundleContext, should we mention these?